src/lib/address_resolve/AddressResolve.h

```diff
@@ -247,6 +247,11 @@ class Resolver
     /// a clear decision if the callback should or should not be invoked.
     virtual CHIP_ERROR CancelLookup(Impl::NodeLookupHandle & handle, FailureCallback cancel_method) = 0;
 
+#if CHIP_DEVICE_ENABLE_CASE_DNS_CACHE
```

Comment:
I wonder if for code cleanliness/maintainability if we maybe want to not ifdef these and only have the ifdef into some cache conditional like "null cache" vs "real cache" implementations. This will eat a bit more flash however it will make the code have less ifdefs/complexity.

@bzbarsky-apple what do you think?